dwave/system/testing.py

```diff
@@ -56,15 +56,25 @@ class MockDWaveSampler(dimod.Sampler, dimod.Structured):
             A list of one number [m] for Pegasus, defaulted as [3].
             A list of two numbers [m,t] for Zephyr, defaulted as [2,4].  
     
-    """
+        parameter_warnings (bool, optional, default=True):
+            The MockSampler is adaptive with respect to ``num_reads``,
+            ``answer_mode`` and ``max_answers`` and ``label`` 
+            parameters. By default ``initial_state`` can also be mocked, if
+            dwave-greedy is installed. All other parameters are ignored and a 
+            warning will be raised by default.
 
+    """
+    # Feature suggestion - add seed as an optional input, to allow reproducibility.
+    
     nodelist = None
     edgelist = None
     properties = None
     parameters = None
-
+    parameter_warnings = True
     def __init__(self, broken_nodes=None, broken_edges=None,
-                 topology_type='chimera',topology_shape=None, **config):
+                 topology_type='chimera',topology_shape=None,
+                 parameter_warnings=True, **config):
+        self.parameter_warnings = parameter_warnings
         
         if topology_type == 'zephyr':
```

Comment:
btw, to @[pau557](https://github.com/pau557)'s point about inheriting `to_networkx_graph()`, I bet you can use that method here

Reply:
This is good practice for sure.
However, to_networkx_graph contains no argument checking, and contains an error in the handling of the zephyr shape parameter (one not two arguments). So I'll correct those as well.